test/token/otcEscrow.spec.ts

```diff
@@ -0,0 +1,340 @@
+
+import { Account } from "@utils/types";
+import { IndexToken, OtcEscrow } from "@utils/contracts";
+import DeployHelper from "@utils/deploys";
+import {
+  addSnapshotBeforeRestoreAfterEach,
+  ether,
+  getAccounts,
+  getLastBlockTimestamp,
+  getProvider,
+  getRandomAccount,
+  getRandomAddress,
+  getWaffleExpect,
+} from "@utils/index";
+import { ContractTransaction } from "ethers";
+import { StandardTokenMock } from "@typechain/StandardTokenMock";
+import { Vesting__factory } from "@typechain/factories/Vesting__factory";
+
+const expect = getWaffleExpect();
+
+describe("OtcEscrow", () => {
+  let owner: Account;
+  let indexGov: Account;
+  let investor: Account;
+  
+  let deployer: DeployHelper;
+  let index: IndexToken;
+  let usdc: StandardTokenMock;
+
+  before(async () => {
+    [
+      owner,
+      indexGov,
+      investor
+    ] = await getAccounts();
+
+    deployer = new DeployHelper(owner.wallet);
+
+    index = await deployer.token.deployIndexToken(owner.address);
+    await index.transfer(indexGov.address, ether(1000));
+    usdc =  await deployer.mocks.deployStandardTokenMock(owner.address);
+    await usdc.transfer(investor.address, BigNumber.from(1_000_000 * 10**6));
+  });
+
+  addSnapshotBeforeRestoreAfterEach();
+
+  describe("#constructor", async () => {
+
+    let subjectVestingStart: BigNumber;
+    let subjectVestingCliff: BigNumber;
+    let subjectVestingEnd: BigNumber;
+    let subjectUSDCAmount: BigNumber;
+    let subjectIndexAmount: BigNumber;
+
+    beforeEach(async () => {
+      const now = await getLastBlockTimestamp()
+      subjectVestingStart = now;
+      subjectVestingCliff = now.add(60 * 60 * 24 * 183);
+      subjectVestingEnd = now.add(60 * 60 * 24 * 547);
+      subjectUSDCAmount = BigNumber.from(100_000 * 10**6);
+      subjectIndexAmount = ether(100);
+    });
+
+    async function subject(): Promise<OtcEscrow> {
+      return await deployer.token.deployOtcEscrow(
+        investor.address,
+        indexGov.address,
+        subjectVestingStart,
+        subjectVestingCliff,
+        subjectVestingEnd,
+        subjectUSDCAmount,
+        subjectIndexAmount,
+        usdc.address,
+        index.address
+      );
+    }
+
+    it("should set the correct state variables", async () => {
+      const escrow = await subject();
+
+      expect(await escrow.beneficiary()).to.eq(investor.address);
+      expect(await escrow.indexGov()).to.eq(indexGov.address);
+      expect(await escrow.vestingStart()).to.eq(subjectVestingStart);
+      expect(await escrow.vestingCliff()).to.eq(subjectVestingCliff);
+      expect(await escrow.vestingEnd()).to.eq(subjectVestingEnd);
+      expect(await escrow.usdcAmount()).to.eq(subjectUSDCAmount);
+      expect(await escrow.indexAmount()).to.eq(subjectIndexAmount);
+      expect(await escrow.usdc()).to.eq(usdc.address);
+      expect(await escrow.index()).to.eq(index.address);
+    });
+  });
+
+  describe("#swap", async () => {
```

Comment:
Let's just make a simple test case here to make sure that `beneficiary` can call as well.